File: src/main/java/rag/answer/AnswerAgent.java
package main.java.rag.answer;

import main.java.rag.data.ChunkStore;
import main.java.rag.model.Answer;
import main.java.rag.model.Hit;

import java.util.List;

/**
 * Strategy interface for answer generation
 * SOLID: DIP - Depend on abstraction
 * Design Pattern: Strategy
 */
public interface AnswerAgent {
    /**
     * Generate answer from top hits
     * @param query The query terms
     * @param topHits The top retrieved and reranked hits
     * @param chunkStore The chunk store for reference
     * @return Answer with text and citations
     */
    Answer answer(List<String> query, List<Hit> topHits, ChunkStore chunkStore);
}

File: iteration2/src/answer/answer_agent.py
from abc import ABC, abstractmethod
from typing import List
from src.model.hit import Hit
from src.model.answer import Answer
from src.data.chunk_store import ChunkStore


class AnswerAgent(ABC):
    @abstractmethod
    def answer(self, query: List[str], top_hits: List[Hit], chunk_store: ChunkStore) -> Answer:
        
        pass
